my-app/src/app/courses-page/services/courses.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+
+import { Course } from '../../models/course.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CoursesService {
+  courses: Course[] = [
+    new Course(1,
+      'Video Course 1. Name tag',
+      new Date('2020/12/01'),
+      150,
+      `Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college's classes. They're published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.`,
+      true),
+    new Course(1,
+      'Video Course 2. Name tag',
+      new Date('2020/10/28'),
+      320,
+      `Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college's classes. They're published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.`,
+      false),
+    new Course(1,
+      'Video Course 3. Name tag',
+      new Date('2020/11/24'),
+      0,
+      `Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college's classes. They're published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.`,
+      false),
+  ];
+
+  constructor() { }
+
+  getCoursesList(): Course[] {
+    return this.courses;
+  }
+
+  createCourseItem(course: Course): void {
+    this.courses.push(course);
+  }
+
+  getCourseById(id: number): Course {
+    return this.courses.find(courseItem => courseItem.id === id);
+  }
+
+  updateCourseItem(course: Course): void {
+    const updatedIndex = this.courses.findIndex(courseItem => courseItem.id === course.id);
+    console.log('updatedIndex==', updatedIndex);
+    this.courses[updatedIndex] = course;
```

Comment:
I would recommend creating new array. Instead of just overriding a specific index. It will be helpful in the future when you will learn about onPush change detection strategy =)